rst_files/lake_model.rst

```diff
@@ -217,60 +217,76 @@ Setup
 
 .. code-block:: julia
 
-    function LakeModel(;λ = 0.283, α = 0.013, b = 0.0124, d = 0.00822)
+    LakeModel = @with_kw (λ = 0.283, α = 0.013, b = 0.0124, d = 0.00822)
+
+    function A_matrices(lm)
+        @unpack λ, α, b, d = lm
         g = b - d
-        A = [(1 - λ) * (1 - d) + b (1 - d) * α + b   # carry out intermediate calculations
-            (1 - d) * λ            (1 - d) * (1 - α)]
+        A = [(1 - λ) * (1 - d) + b      (1 - d) * α + b
+            (1 - d) * λ                 (1 - d) * (1 - α)]
         A_hat = A ./ (1 + g)
-        return (λ = λ, α = α, b = b, d = d, g = g, A = A, A_hat = A_hat) # return a NamedTuple
+        return (g = g, A = A, A_hat = A_hat)
     end
 
     function rate_steady_state(lm)
-        sol = fixedpoint(x -> lm.A_hat * x, fill(0.5, 2))
+        A_hat = A_matrices(lm).A_hat
+        sol = fixedpoint(x -> A_hat * x, fill(0.5, 2))
         converged(sol) || error("Failed to converge in $(result.iterations) iterations")
         return sol.zero
     end
 
     function simulate_stock_path(lm, X0, T)
+        A = A_matrices(lm).A
         X_path = zeros(eltype(X0), 2, T)
         X = copy(X0)
         for t in 1:T
             X_path[:, t] = X
-            X = lm.A * X
+            X = A * X
         end
         return X_path
     end
 
     function simulate_rate_path(lm, x0, T)
+        A_hat = A_matrices(lm).A_hat
```

Comment:
```
@unpack A_hat = A_matrices(lm)
```
for example....

Reply:
Done. I just felt weird using `@unpack` for one variable.